x-pack/plugins/lens/public/app_plugin/types.ts

```diff
@@ -120,6 +121,8 @@ export interface LensTopNavMenuProps {
   theme$: Observable<CoreTheme>;
   indexPatternService: IndexPatternServiceAPI;
   onTextBasedSavedAndExit: ({ onSave }: { onSave: () => void }) => Promise<void>;
+  shortUrlService?: (params: LensAppLocatorParams) => Promise<string>;
```

Comment:
I don't think there is a condition where this service isn't available. I made it required both here and in the functions in `share_action.ts` and ran a type check. It passed just fine.

```suggestion
  shortUrlService: (params: LensAppLocatorParams) => Promise<string>;
```

Reply:
True, the `share` plugin is required 👍